src/components/AddApplication/AddApplicationSchema.js

```diff
@@ -91,10 +198,18 @@ const fields = (applications = [], intl, sourceTypes, applicationTypes, authenti
         }
 
         nextStep = ({ values: { application } }) => {
-            const app = application ? application : {};
-            const appId = app.application_type_id ? app.application_type_id : '';
+            if (application) {
+                const appType = applicationTypes.find(({ id }) => id === application.application_type_id);
```

Comment:
You are not using `appType` anywhere, right? Might be a good idea to use the function directly
```JS
if (hasAlreadySupportedAuthType(
  authenticationValues,
  applicationTypes.find(({ id }) => id === application.application_type_id),
  sourceType.name
)) {
   return 'selectAuthentication';
}
```